.github/workflows/PullRequestAndMergeMaster.yml

```diff
@@ -0,0 +1,80 @@
+name: PR/Release pipeline
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+    branches:
+  release:
+    types:
+      - created
+jobs:
+  BuildAndTest:
+    name: Build And Test
+    runs-on: ubuntu-latest
+    env:
+      GOPATH: ${{ github.workspace }}
+      GO111MODULE: off
+    defaults:
+      run:
+        working-directory: ${{ env.GOPATH }}/src/github.com/${{ github.repository }}
+
+    steps:
+    - uses: actions/setup-go@v2
+      with:
+        go-version: 1.13
+      id: go
+
+    - name: Checkout code
+      uses: actions/checkout@v2
+      with:
+        path: ${{ env.GOPATH }}/src/github.com/${{ github.repository }}
+
+    - name: Install dependencies 
+      run: |
+        go get -u -v github.com/golang/dep/cmd/dep
+
+        sudo apt update && sudo apt install -y build-essential zlibc zlib1g-dev ruby ruby-dev openssl libxslt1-dev libxml2-dev libssl-dev libreadline-dev libyaml-dev libsqlite3-dev sqlite3
+        
+        curl -O -L https://github.com/cloudfoundry/bosh-cli/releases/download/v6.2.0/bosh-cli-6.2.0-linux-amd64
+        chmod +x ./bosh-cli-6.2.0-linux-amd64
+        sudo mv ./bosh-cli-6.2.0-linux-amd64 /usr/local/bin/bosh
+        
+        curl -O -L https://github.com/cf-platform-eng/tile-generator/releases/download/v14.0.3/tile_linux-64bit
+        chmod +x tile_linux-64bit
+        sudo mv tile_linux-64bit /usr/local/bin/tile
+
+    - name: Get version
+      id: get_version
+      if: github.event.release.tag_name != ''
+      run: |
+        TAG=${{ github.event.release.tag_name }}
+        RELEASE_TAG=${TAG//v}
+        echo "RELEASE_TAG=${RELEASE_TAG}" >> $GITHUB_ENV
+        echo "::set-output name=VERSION::${RELEASE_TAG}"
+
+    - name: Test and Build
+      run: |
+        make release
+
+    - name: Upload Tile
+      if: github.event.release.tag_name != ''
```

Comment:
why are we checking  the tag_name here? I don't see it being used.

Reply:
in order to avoid re-building when released, i use this condition to use the generated assets when a release is created (that should have a tag)

Reply:
also this step uses `VERSION` that is produced in `get_version` that needs the tag.

Reply:
Ok. If you change the release types to released and prereleased it always has a tag so you don't to check for it.
To avoid re-building don't know you can do that easily. Maybe check if there are already assets in the pre-release?